libraries/stdlib/test/text/RegexTest.kt

```diff
@@ -320,7 +320,14 @@ class RegexTest {
     @Test fun replaceEvaluator() {
         val input = "/12/456/7890/"
         val pattern = "\\d+".toRegex()
-        assertEquals("/2/3/4/", pattern.replace(input, { it.value.length.toString() }))
+        assertEquals("/2/3/4/", pattern.replace(input) { it.value.length.toString() })
+    }
+
+    @Test fun replaceFirstEvaluator() {
+        val pattern = "\\d+".toRegex()
+        assertEquals("no_match", pattern.replaceFirst("no_match") { it.value.length.toString() })
+        assertEquals("6", pattern.replaceFirst("012345") { it.value.length.toString() })
+        assertEquals("abc[123]de", pattern.replaceFirst("abc123de") { "[${it.value}]" })
```

Comment:
Makes sense to check an input string of "abc123de56" to ensure that only the _first_ match is actually replaced.